### PR TITLE
feat: add description value in ibm_iam_authorization_policy block

### PR DESCRIPTION
**Pull request by Khuzaima05**
### Description

Pass description value to ibm_iam_authorization_policy block.

issue : https://github.ibm.com/GoldenEye/issues/issues/5662

### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [x] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### Merge actions for mergers

- When merging, use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents and any release notes provided by the PR author. The commit message determines whether a new version of the module is needed, and if so, which semver increment to use (major, minor, or patch).
- Merge by using "Squash and merge".

### CODE REVIEW

**Comment by Khuzaima05** [comment]
@ocofaigh Can I skip the upgrade test here, Pipeline is failing due to this.

**Comment by ocofaigh** [comment]
@Khuzaima05 yes I think this is fine. As per the log, its going to simply update the description of the existing auth policy, so no impact to consumers..

```
Messages:   	Resource(s) identified to be updated Name: kms_policy Address: module.postgresql_db.ibm_iam_authorization_policy.kms_policy[0] Actions: [update]
        	            	DIFF:
        	            	Before: {"description":null}
        	            	After: {"description":"Allow all ICD Postgres instances in the resource group ee8675e51c5d4ef5aec89b49c5a1b7b5 to read from the kms instance GUID 3082d097-48e5-4adb-b613-77e348d40c89"}
```

File: main.tf [hunk 1]
```diff
@@ -36,6 +36,7 @@ resource "ibm_iam_authorization_policy" "kms_policy" {
   target_service_name         = local.kms_service
   target_resource_instance_id = var.existing_kms_instance_guid
   roles                       = ["Reader"]
+  description                 = "Allow all ICD Postgres instances in the resource group ${var.resource_group_id}  to read from the ${local.kms_service} instance GUID ${var.existing_kms_instance_guid} "
```
Comment (ocofaigh) [inline main.tf:39]: remove the double space after `${var.resource_group_id}`

**Review by ocofaigh — CHANGES_REQUESTED** [review]
one minor comment

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**